jina/peapods/runtimes/gateway/graph/topology_graph.py

```diff
@@ -0,0 +1,135 @@
+import asyncio
+
+from collections import defaultdict
+from typing import List, Optional, Dict, Tuple
+
+
+class TopologyGraph:
+    """
+    :class TopologyGraph is a class that describes a computational graph of nodes, where each node represents
+        a Pod that needs to be sent messages in the order respecting the path traversal.
+
+    :param graph_description: A dictionary describing the topology of the Pods. 2 special nodes are expected, the name `start-gateway` and `end-gateway` to
+        determine the nodes that receive the very first message and the ones whose response needs to be sent back to the client. All the nodes with no outgoing nodes
+        will be considered to be hanging, and they will be "flagged" so that the user can ignore their tasks and not await them.
+    """
+
+    class _ReqReplyNode:
```

Comment:
The class is part of the interface of the graph as its returned by `origin_nodes`, should it really be private then?

Reply:
Good point